lib/config/hadeployconfig.go

```diff
@@ -0,0 +1,186 @@
+	External      *ExternalSettings      `toml:"external,omitempty"`
+}
+
+type Architecture struct {
+	ExistingInfra *ConfigInitials `toml:"existing_infra,omitempty"`
+	Aws           *ConfigInitials `toml:"aws,omitempty"`
+}
+
+type ConfigInitials struct {
+	SSHUser                     string `toml:"ssh_user,omitempty"`
+	SSHGroupName                string `toml:"ssh_group_name,omitempty"`
+	SSHKeyFile                  string `toml:"ssh_key_file,omitempty"`
+	SSHPort                     string `toml:"ssh_port,omitempty"`
+	SecretsKeyFile              string `toml:"secrets_key_file,omitempty"`
+	SecretsStoreFile            string `toml:"secrets_store_file,omitempty"`
+	SudoPassword                string `toml:"sudo_password,omitempty"`
+	LoggingMonitoringManagement string `toml:"logging_monitoring_management,omitempty"`
+	Architecture                string `toml:"architecture,omitempty"`
+	WorkspacePath               string `toml:"workspace_path,omitempty"`
+	BackupMount                 string `toml:"backup_mount,omitempty"`
+	BackupConfig                string `toml:"backup_config,omitempty"`
+	S3BucketName                string `toml:"s3_bucketName,omitempty"`
+	HabitatUIDGid               string `toml:"habitat_uid_gid,omitempty"`
+	// Elk configs not being used, it's not added
+}
+
+type ObjectStorage struct {
+	Config *ConfigObjectStorage `toml:"config,omitempty"`
+}
+
+type CertByIP struct {
+	IP         string `toml:"ip,omitempty"`
+	PrivateKey string `toml:"private_key,omitempty"`
+	PublicKey  string `toml:"public_key,omitempty"`
+	NodesDn    string `toml:"nodes_dn,omitempty"`
+}
+
+type ConfigObjectStorage struct {
+	BucketName string `toml:"bucket_name,omitempty"`
+	AccessKey  string `toml:"access_key,omitempty"`
+	SecretKey  string `toml:"secret_key,omitempty"`
+	Endpoint   string `toml:"endpoint,omitempty"`
+	Region     string `toml:"region,omitempty"`
+}
+
+type AutomateSettings struct {
+	Config *ConfigAutomateSettings `toml:"config,omitempty"`
+}
+type ConfigAutomateSettings struct {
+	AdminPassword string `toml:"admin_password,omitempty"`
+	Fqdn          string `toml:"fqdn,omitempty"`
+	ConfigFile    string `toml:"config_file,omitempty"`
+	TeamsPort     string `toml:"teams_port,omitempty"`
+	ConfigSettings
+}
+
+type ServerConfigSettings struct {
+	Config *ConfigSettings `toml:"config,omitempty"`
+}
+
+type ConfigSettings struct {
+	InstanceCount     string      `toml:"instance_count,omitempty"`
+	EnableCustomCerts bool        `toml:"enable_custom_certs,omitempty"`
+	RootCA            string      `toml:"root_ca,omitempty"`
+	PrivateKey        string      `toml:"private_key,omitempty"`
+	PublicKey         string      `toml:"public_key,omitempty"`
+	AdminCert         string      `toml:"admin_cert,omitempty"`
+	AdminKey          string      `toml:"admin_key,omitempty"`
+	AdminDn           string      `toml:"admin_dn,omitempty"`
+	NodesDn           string      `toml:"nodes_dn,omitempty"`
```

Comment:
I think these also are only needed for OpenSearch, we can create separate Struct and use for OS settings